flipperdevices/flipperzero-firmware
# add formatting to DESfire data dump

Previously, the binary data for each record on a DESfire file is printed out as a long hex string, with no delineation between characters. This makes it hard to skim for e.g. ascii data.

This change adds some formatting code to output the binary data as a mini hex dump, complete with offsets and ascii printing. Due to the screen size, each row includes only 4 bytes of data.

Before:
```
File 15
standard plain
r 14 w 15 rw 15 c 2
size 9
0004b55500d8399008
```

After:
```
File 15
standard plain
r 14 w 15 rw 15 c 2
size 9
record 0
000|00 04 b5 55 ...U
004|00 d8 39 90 ..9.
008|08          .
```

# What's new

- Adds hexdump formatting for DESfire file data

# Verification 

1. Read a DESfire tag or load a saved file from a DESfire that contains at least one file
2. Browse to the tag info, and choose one of the files
3. Observe the formatted hex dump of the file

# Checklist (For Reviewer)
